src/memory/client/MemoryBrowser.tsx

```diff
@@ -101,14 +130,26 @@ export class MemoryBrowser extends React.Component<Props, State> {
           error: undefined,
           memory: undefined,
         });
-        const result = await messageBroker.send({
-          command: 'ReadMemory',
-          args: {
-            address: this.addressReq,
-            length: parseInt(this.lengthReq),
-          },
-        });
-        this.setState({ memory: result.result });
+        if (this.state.childrenNames.length > 0) {
+          const result = await messageBroker.send({
+            command: 'ReadMemory',
+            args: {
+              address: this.addressReq,
+              length: parseInt(this.lengthReq),
+              child: this.childReq,
+            },
+          });
+          this.setState({ memory: result.result });
+        } else {
+          const result = await messageBroker.send({
+            command: 'ReadMemory',
+            args: {
+              address: this.addressReq,
+              length: parseInt(this.lengthReq),
+            },
+          });
+          this.setState({ memory: result.result });
+        }
```

Comment:
```suggestion
        const result = await messageBroker.send({
          command: 'ReadMemory',
          args: {
            address: this.addressReq,
            length: parseInt(this.lengthReq),
            child: this.state.childrenNames.length ? this.childReq : undefined,
          },
        });
        this.setState({ memory: result.result });
```

Reply:
I have updated it.

Reply:
Resolving here, but I don't see the change. I will make a new comment in the updated code.